src/components/Auth.vue

```diff
@@ -112,6 +112,19 @@ export default {
     if (this.$cookies.isKey('newRoads')) {
       const newRoads = this.$cookies.get('newRoads');
       if (Object.keys(newRoads).length) {
+        for (var roadID in newRoads) {
+          if (!Array.isArray(newRoads[roadID].contents.selectedSubjects[0])) {
+            const simpless = Array.from(Array(16), () => new Array());
+            for (let i = 0; i < newRoads[roadID].contents.selectedSubjects.length; i++) {
+              const s = newRoads[roadID].contents.selectedSubjects[i];
+              if (s.semester === undefined || s.semester < 0) {
+                s.semester = 0;
+              }
+              simpless[s.semester].push(s);
+            }
+            newRoads[roadID].contents.selectedSubjects = simpless;
+          }
+        }
```

Comment:
This converts the flat list of subjects fireroad has into a list of lists broken down by semester, which is easier/faster for our system to handle.  I'm pretty sure it will break fireroad to send the converted data back.